libraries/AP_Beacon/AP_Beacon.cpp

```diff
@@ -214,7 +214,7 @@ bool AP_Beacon::beacon_healthy(uint8_t beacon_instance) const
 // return distance to beacon in meters
 float AP_Beacon::beacon_distance(uint8_t beacon_instance) const
 {
-    if (!beacon_state[beacon_instance].healthy || beacon_instance >= num_beacons) {
+    if ( beacon_instance >= num_beacons || !beacon_state[beacon_instance].healthy) {
```

Comment:
```suggestion
    if (beacon_instance >= num_beacons || !beacon_state[beacon_instance].healthy) {
```

Even better if it was two lines IMO - might not have had the bug in the first place then.